File: app/src/main/java/com/ruzhan/rxrepository/db/converter/RoomDataConverter.java
package com.ruzhan.rxrepository.db.converter;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 * Created by ruzhan123 on 2018/6/21.
 */
public class RoomDataConverter {

    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/ruzhan/rxrepository/db/converter/RoomDataConverter.kt
package com.ruzhan.rxrepository.db.converter


import androidx.room.TypeConverter
import java.util.*

object RoomDataConverter {

    @TypeConverter
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }
}
